### ISSUE TITLE
Bug: Querying Date attributes fail when sharding

### ISSUE THREAD

**Issue by zoltan-fedor** [semi-technologies/weaviate#1775]
When running the latest Weaviate (v1.9.0) in sharded mode (eg multiple replicas in Kubernetes), then filtering with date attributes will result in error `expected value to be time.Time, got string`.
When Weaviate is redeployed with a single shard, the error goes away.

See more about this being discussed under thread: https://app.slack.com/client/T0181DYT9KN/C017EG2SL3H/thread/C017EG2SL3H-1640380914.476400

The query and the resulting error:
```
{
    Get {
        HotTopics(
            nearText: {
                concepts: ["whatever"]
            }
            limit: 4
            where:{ 
                operator:And
                operands: [{
                    operator: LessThan
                    path: ["createddate"]
                    valueDate: "2021-12-01T00:00:00Z"
                }, {
                    operator: GreaterThan
                    path: ["closeddate"]
                    valueDate: "2021-12-10T00:00:00Z"
                }]
            }
        ) {
          createddate
          closeddate
          _additional {
            id
            certainty
          }
        }
    }
}

    >>>>
    'errors': [{'locations': [{'column': 9, 'line': 4}],
                 'message': 'explorer: get class: vector search: object vector '
                            'search at index hottopics: remote shard BiuJ0de5fLih: '
                            'unexpected status code 500 (shard '
                            'hottopics_BiuJ0de5fLih: build inverted filter allow '
                            'list: nested clause at pos 0: expected value to be '
                            'time.Time, got string\n'
                            ')',
```

The given attributes are defined and uploaded as date properly - see below.
```
-- attribute definitions are properly set to 'date':
        {
            "name": "createddate",
            "dataType": ["date"],
            "description": "When the HT was created",
            'moduleConfig': {
                'text2vec-transformers': {
                    'skip': True,
                    'vectorizePropertyName': False
                }
            },
        },
        {
            "name": "closeddate",
            "dataType": ["date"],
            "description": "When the HT was created",
            'moduleConfig': {
                'text2vec-transformers': {
                    'skip': True,
                    'vectorizePropertyName': False
                }
            },
        }

-- objects are uploaded as:
        'data_object': {
            'createddate': ht['CREATEDDATE'].strftime('%Y-%m-%dT%H:%M:%SZ'),  # 2021-10-02T15:00:00Z
            'closeddate': ht['CLOSEDDATE'].strftime('%Y-%m-%dT%H:%M:%SZ')
        }
```

I have enabled debug logging on both of the weaviate shards, unfortunately the logging was still quite limited during the query throwing the error - see logs below.

Debug log from instance 0 ('weaviate-0'):
```
{"level":"debug","msg":" memberlist: Stream connection from=10.47.119.190:60954","time":"2021-12-29T16:04:58Z"}
{"level":"debug","msg":" memberlist: Initiating push/pull sync with: weaviate-1 10.47.119.190:7000","time":"2021-12-29T16:05:07Z"}
{"level":"debug","msg":" memberlist: Stream connection from=10.47.119.190:33210","time":"2021-12-29T16:05:13Z"}
{"action":"restapi_request","level":"debug","method":"POST","msg":"received HTTP request","time":"2021-12-29T16:05:20Z","url":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/v1/graphql","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":"","RawFragment":""}}
{"level":"debug","msg":" memberlist: Initiating push/pull sync with: weaviate-1 10.47.119.190:7000","time":"2021-12-29T16:05:22Z"}
```
Debug log from instance 1 ('weaviate-1'):
```
{"level":"debug","msg":" memberlist: Initiating push/pull sync with: weaviate-0 10.47.112.97:7000","time":"2021-12-29T16:04:43Z"}
{"level":"debug","msg":" memberlist: Stream connection from=10.47.112.97:49216","time":"2021-12-29T16:04:52Z"}
{"level":"debug","msg":" memberlist: Initiating push/pull sync with: weaviate-0 10.47.112.97:7000","time":"2021-12-29T16:04:58Z"}
{"action":"restapi_request","level":"debug","method":"POST","msg":"received HTTP request","time":"2021-12-29T16:05:05Z","url":{"Scheme":"","Opaque":"","User":null,"Host":"","Path":"/v1/graphql","RawPath":"","ForceQuery":false,"RawQuery":"","Fragment":"","RawFragment":""}}
{"level":"debug","msg":" memberlist: Stream connection from=10.47.112.97:49578","time":"2021-12-29T16:05:07Z"}
{"level":"debug","msg":" memberlist: Initiating push/pull sync with: weaviate-0 10.47.112.97:7000","time":"2021-12-29T16:05:13Z"}
{"level":"debug","msg":" memberlist: Stream connection from=10.47.112.97:49988","time":"2021-12-29T16:05:22Z"}
{"level":"debug","msg":" memberlist: Initiating push/pull sync with: weaviate-0 10.47.112.97:7000","time":"2021-12-29T16:05:28Z"}
{"level":"debug","msg":" memberlist: Stream connection from=10.47.112.97:50384","time":"2021-12-29T16:05:37Z"}
```

**Comment by etiennedi** [comment]
FYI: The investigation for this highlighted a second, related bug:

There is also a bug wrt importing items with date fields in a distributed setting, it seems that when an object is sent to a remote shard, we lose the type information about this being a timestamp. As a result the timestamp is indexed as a string, making it impossible to find using filters - even with the original bug fixed.

Will also investigate and fix the second bug.

**Comment by etiennedi** [comment]
closed in #1786
